doc/Programming-guide/Key-platform-features/Devices-provisioning-and-registration/index.md

```diff
@@ -9,77 +9,75 @@ sort_idx: 10
-and then with the Operations service to submit the endpoint data and complete the registration.
+## Endpoint registration
 
+To function within a [Kaa cluster]({{root_url}}Glossary/#kaa-cluster), every endpoint must be registered using the security credentials and the endpoint profile.
+During the registration process, the endpoint first communicates with one or multiple [Bootstrap services]({{root_url}}Glossary/#bootstrap-service) to obtain a list of available [Operations services]({{root_url}}Glossary/#operations-service).
+Then it communicates with the Operations service to submit the endpoint data and complete the registration.
+
+The endpoint registration process is illustrated in the picture below.
 
 ![Endpoint Registration](attach/registration.png)
 
-### Operations service list resolution
+### Operations services list
 
-Initially, the Kaa endpoint SDK does not include any information about Operations services for security and load-balancing reasons. 
-Instead, during the SDK generation, a Control service embeds a list of available Bootstrap services into the SDK 
-(using a properties file for Java implementation, a header file for C++, etc.). 
-The endpoint selects a random Bootstrap service from the list and sends a resolution request to the selected Bootstrap service. 
-The resolution request contains the application token that can be used by the Bootstrap service in specific load-balancing strategies. 
-The resolution response from the Bootstrap service contains a prioritized list of Operations services with their connectivity details and 
-is signed with the Bootstrap service private key. 
-The endpoint verifies the signature of the response by using the Bootstrap public key available from the build time parameters and retains the received information.
+For security and load-balancing reasons, Kaa endpoint SDKs do not include any information about Operations services.
+Instead, a [Control service]({{root_url}}Glossary/#control-service) embeds a list of available Bootstrap services into the SDK during the SDK generation process.
+This list is saved as a properties file in case with Java SDK, and as a header file in case with other [SDK types]({{root_url}}Glossary/#sdk-type).
+The endpoint selects a random Bootstrap service from the list and sends a resolution request to the selected Bootstrap service.
+The resolution request contains the [application token]({{root_url}}Glossary/#application-token) that can be used by the Bootstrap service in specific load-balancing strategies.
+The resolution response from the Bootstrap service contains a prioritized list of Operations services with their connectivity details and is signed with the Bootstrap service private key.
+The endpoint verifies the signature of the response using the Bootstrap public key available from the build time parameters and saves the received information.
 
-### Endpoint Authorization
+### Endpoint authorization
```

Comment:
After another look, this is not even authentication. Authentication occurs only **after** the endpoint is registered. This section in fact describes credentials verification prior to the registration.